tower_modules/tower_inventory/tasks/main.yml

```diff
@@ -31,4 +73,4 @@
 
 - assert:
     that:
-      - "result.msg =='Failed to update inventory, organization not found: The requested object could not be found.'"
\ No newline at end of file
+      - "result.msg =='The organizations test-non-existing-org was not found on the Tower server'"
```

Comment:
If we could make this match the old message, then we would have a good chance at merging this PR. Otherwise, testing with multiple versions gives headaches (because this repo is not versioned). Ping @squidboylan 

The other changes here look like expansion of test coverage, which doesn't strike me as posing a problem.

Reply:
You can still change this assertion to be _less_ restrictive and accommodate a new message.